streams/src/main/java/org/apache/kafka/streams/processor/internals/StateRestorer.java

```diff
@@ -51,14 +57,30 @@ long checkpoint() {
         return checkpoint == null ? NO_CHECKPOINT : checkpoint;
     }
 
-    void restore(final byte[] key, final byte[] value) {
-        stateRestoreCallback.restore(key, value);
+    void notifyRestoreStarted(long startingOffset, long endingOffset) {
```

Comment:
nit: rename this function to `restoreStarted` to be consistent with other names. Such will help other code readers to understand these functions are for the same code granularity and semantics.